sql/logictest: TestLogic failed

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1856998&tab=buildLog) on [release-20.1@7699136a4861a6f6423e16ab7954fc4ac226bd6a](https://github.com/cockroachdb/cockroach/commits/7699136a4861a6f6423e16ab7954fc4ac226bd6a):

```
=== RUN   TestLogic
--- FAIL: TestLogic (1094.19s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestLogic188802467
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestLogic/fakedist-vec/vectorize_types
=== PAUSE TestLogic/fakedist-vec/vectorize_types
=== CONT  TestLogic/fakedist-vec/vectorize_types
--- done: testdata/logic_test/with with config fakedist-vec: 67 tests, 0 failures
rewrote:
CREATE TABLE o (gf INT8 REFERENCES t (g), h INT8, i INT8, INDEX ii (i) STORING (h), FAMILY (i, gf, h));


        --- FAIL: TestLogic/fakedist-vec/vectorize_types (0.98s)
            logic.go:1564: 
                 
                testdata/logic_test/vectorize_types:82: 
                expected:
                pq: unable to vectorize execution plan: unhandled type int\[\]
                
                got:
                pq: unsupported type int[]
            logic.go:2409: 
                testdata/logic_test/vectorize_types:85: error while processing
            logic.go:2409: testdata/logic_test/vectorize_types:85: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-vec
    --- FAIL: TestLogic/fakedist-vec (0.47s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46713 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master)

- #46424 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003231705_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003231705_v20.1.0-beta.3)

- #46268 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003181957_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003181957_v20.1.0-beta.3)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1857155&tab=buildLog) on [release-20.1@7699136a4861a6f6423e16ab7954fc4ac226bd6a](https://github.com/cockroachdb/cockroach/commits/7699136a4861a6f6423e16ab7954fc4ac226bd6a):

```
=== RUN   TestLogic
--- FAIL: TestLogic (1107.49s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestLogic972113089
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestLogic/fakedist-vec/vectorize_types
=== PAUSE TestLogic/fakedist-vec/vectorize_types
=== CONT  TestLogic/fakedist-vec/vectorize_types
rewrote:
CREATE TABLE child (id INT8 PRIMARY KEY, parent_id INT8 UNIQUE REFERENCES parent, FAMILY (parent_id), FAMILY (id));


        --- FAIL: TestLogic/fakedist-vec/vectorize_types (0.58s)
            logic.go:1564: 
                 
                testdata/logic_test/vectorize_types:82: 
                expected:
                pq: unable to vectorize execution plan: unhandled type int\[\]
                
                got:
                pq: unsupported type int[]
            logic.go:2409: 
                testdata/logic_test/vectorize_types:85: error while processing
            logic.go:2409: testdata/logic_test/vectorize_types:85: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-vec
    --- FAIL: TestLogic/fakedist-vec (0.36s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46713 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master)

- #46424 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003231705_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003231705_v20.1.0-beta.3)

- #46268 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003181957_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003181957_v20.1.0-beta.3)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(sql/logictest).TestLogic failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=1859564&tab=buildLog) on [release-20.1@a84b18f04688cfb01a0487a5792bf94c0f74d061](https://github.com/cockroachdb/cockroach/commits/a84b18f04688cfb01a0487a5792bf94c0f74d061):

```
=== RUN   TestLogic
--- FAIL: TestLogic (1064.80s)
    test_log_scope.go:77: test logs captured to: /tmp/logTestLogic513317019
    test_log_scope.go:58: use -show-logs to present logs inline
=== RUN   TestLogic/fakedist-vec/vectorize_types
=== PAUSE TestLogic/fakedist-vec/vectorize_types
=== CONT  TestLogic/fakedist-vec/vectorize_types
        --- FAIL: TestLogic/fakedist-vec/vectorize_types (0.90s)
            logic.go:1564: 
                 
                testdata/logic_test/vectorize_types:82: 
                expected:
                pq: unable to vectorize execution plan: unhandled type int\[\]
                
                got:
                pq: unsupported type int[]
            logic.go:2409: 
                testdata/logic_test/vectorize_types:85: error while processing
            logic.go:2409: testdata/logic_test/vectorize_types:85: too many errors encountered, skipping the rest of the input
=== RUN   TestLogic/fakedist-vec
    --- FAIL: TestLogic/fakedist-vec (0.31s)

```

<details><summary>More</summary><p>
Parameters:

- GOFLAGS=-json

```
make stressrace TESTS=TestLogic PKG=./pkg/sql/logictest TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```

Related:
- #46713 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-master](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-master)

- #46424 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003231705_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003231705_v20.1.0-beta.3)

- #46268 sql/logictest: TestLogic failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202003181957_v20.1.0-beta.3](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202003181957_v20.1.0-beta.3)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestLogic.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/sql/logictest/testdata/logic_test/vectorize_types
+++ b/pkg/sql/logictest/testdata/logic_test/vectorize_types
@@ -79,7 +79,7 @@ INSERT INTO skip_unneeded_cols VALUES ('63616665-6630-3064-6465-616462656562', 1
 statement ok
 SET vectorize=experimental_always
 
-statement error pq: unable to vectorize execution plan: unhandled type int\[\]
+statement error .*(unhandled|unsupported) type int\[\]
 SELECT _unsupported1 FROM skip_unneeded_cols
 
 query IBB

